### PR TITLE
[Login] Fix issue where Application Passwords disabled screen is not shown

### LINKED ISSUE #12647: Notification for disabled Application Password error with support options
The Android app does not notify users when they face a "Disabled Application Password" situation during login, unlike the iOS app https://github.com/woocommerce/woocommerce-ios/issues/14004. Additionally, the app does not offer a contact support button or guide users to troubleshooting documentation.

More here: peaMlT-U4

**Steps to Reproduce:**
1. Attempt to log in with a disabled application password.
2. Observe the lack of notification and support options.

**Expected Behavior:**
The app should notify users about the "Disabled Application Password" error, provide a "Contact Support" button, and include a link to troubleshooting documentation.

**Actual Behavior:**
No notification or support options are provided.

**Proposed Solution:**
- Implement a notification feature to inform users about the "Disabled Application Password" error.
- Enhance the notification to include more detailed information about the issue.
- Add a "Contact Support" button to the error notification.
- Include a link to troubleshooting documentation in the notification.

screenshot-3

### PR DESCRIPTION
**Pull request by hichamboushaba**
Closes: #12647 

### Description
This PR adds a fix for the above issue: when a site has something that blocks the native login (a custom login page or captcha...) and Application Passwords disabled, we don't show the correct error screen, instead we try to load a WebView with an empty URL, which is confusing to the users.
The fix is to check whether the authorizationUrl is present before proceeding to showing the WebView.

**Note:** I will create a separate PR to simplify the logic of the ViewModel, I noticed that the WebView logic is handled now in a separate screen, and the logic on the current ViewModel is not used, just adds to the complexity of the logic.

### Steps to reproduce
1. Install [WPS Hide Login](https://wordpress.org/plugins/wps-hide-login/) plugin.
2. Install [Disable Application Passwords](https://wordpress.org/plugins/disable-application-passwords/)
3. Open the app and attempt site credentials login.

### Testing information
Confirm that the "application passwords disabled" error screen is shown correctly.

### The tests that have been performed
The above on `trunk` and on this branch.

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/sitecredentials/LoginSiteCredentialsViewModel.kt [hunk 1]
```diff
@@ -308,10 +309,16 @@ class LoginSiteCredentialsViewModel @Inject constructor(
...
+                    if (site.fullAuthorizationUrl.isNotNullOrEmpty()) {
+                        triggerEvent(
+                            ShowApplicationPasswordTutorialScreen(
+                                url = site.applicationPasswordsAuthorizeUrl,
+                                errorMessage = errorMessage
+                            )
+                        )
+                    } else {
+                        triggerEvent(ShowApplicationPasswordsUnavailableScreen(siteAddress, site.isJetpackConnected))
+                    }
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/sitecredentials/LoginSiteCredentialsViewModel.kt]: This is the main change of the PR

**Comment by irfano** [comment]
|trunk|PR|
|-|-|
|||

@hichamboushaba, these are screenshots from my testing. In `trunk` testing, it opens a warning screen with a "Continue" button and the button opens an empty webview. In my PR testing, it shows a warning message related to the Application password in my PR test. I’m sharing this to confirm if this is the expected behavior.

**Review by irfano — APPROVED** [review]
LGTM! 👍🏻  Thanks for the fix.

**Comment by hichamboushaba** [comment]
> I’m sharing this to confirm if this is the expected behavior.

Thanks @irfano for the review, that's the expected behavior.

**Pull request merged**